internal/matrix/database/matrix_queries.go

```diff
@@ -1,28 +1,76 @@
 package database
 
+import "fmt"
+
 const (
-	createMatrix    = "create matrix"
-	deleteMatrix    = "delete matrix by uuid"
-	getMatrix       = "get matrix by uuid"
-	getCourseMatrix = "get course and matrix by uuid"
-	listMatrix      = "list matrices"
-	updateMatrix    = "update matrix by uuid"
-	addSubject      = "adds subject to matrix"
-	removeSubject   = "remove subject from matrix"
+	returningMatrixColumns = `matrices.uuid, matrices.code, matrices.name, matrices.description,
+		matrices.created_at, matrices.updated_at`
+	// CREATE.
+	createMatrix = "create matrix"
+	addSubject   = "adds subject to matrix"
+	// READ.
+	getMatrix             = "get matrix by uuid"
+	getCourseMatrixExists = "get whether there is a relationship between course and matrix"
+	listMatrices          = "list matrices"
+	listCourseMatrices    = "list course matrices by course_uuid"
+	// UPDATE.
+	updateMatrix = "update matrix by uuid"
+	// DELETE.
+	deleteMatrix  = "delete matrix by uuid"
+	removeSubject = "remove subject from matrix"
 )
 
 func queriesMatrix() map[string]string {
 	return map[string]string{
-		createMatrix: "INSERT INTO matrices (name, description) VALUES ($1, $2) RETURNING *",
-		deleteMatrix: "UPDATE matrices SET deleted_at = NOW() WHERE uuid = $1",
-		getMatrix:    "SELECT * FROM matrices WHERE uuid = $1",
-		getCourseMatrix: `SELECT *
+		// CREATE.
+		createMatrix: fmt.Sprintf(`INSERT INTO matrices (
+				course_id, code, name, description
+			) SELECT courses.id, $2, $3, $4
+				FROM courses
+			WHERE courses.uuid = $1 AND courses.deleted_at IS NULL
+			RETURNING %s`, returningMatrixColumns),
+		addSubject: "INSERT INTO matrix_subjects (matrix_id, subject_id) VALUES($1, $2)",
+		// READ.
+		getMatrix: fmt.Sprintf(`SELECT
+				courses.uuid AS course_uuid,
+				%s
+			FROM matrices
+			LEFT JOIN courses ON matrices.course_id = courses.id
+			WHERE matrices.uuid = $1 AND matrices.deleted_at IS NULL
+				AND courses.deleted_at IS NULL`, returningMatrixColumns),
+		getCourseMatrixExists: `SELECT
```

Comment:
you could use `EXISTS` here to check if a relationship exists
```
select EXISTS(
 select 
   matrices.id 
 from matrices 
 inner join courses 
   ON matrices.course_id = courses.id
   and courses.deleted_at IS NULL
where matrices.deleted_at IS NULL AND matrices.uuid = $2 and courses.uuid = $1
)
```

Reply:
https://github.com/sumelms/microservice-course/pull/25/commits/c2ae48abf166b91812618e7ba321b433419f2bef